mesos/cli/cli.py

```diff
@@ -35,7 +35,7 @@ def init(arg):
     # arg is:
     # @cli.init         : the function to be decorated
     # @cli.init(parser) : the parsing function
-    cmd_args = None if callable(arg) else arg.parse_args()
+    is_parser = not callable(arg)
```

Comment:
Tests weren't running correctly before.  This fixes it.

parse_args() tries to parse the cli params of the current process.  When run from a mesos-cli program, like mesos-cat, this is fine.  But when run from the tests, the current process is "nosetests".  The tests do mock out the cli params, but not until after they import files which call this decorator:

https://github.com/mesosphere/mesos-cli/blob/master/tests/integration/test_cat.py#L22

This fix makes it so args aren't parsed until the decorated function is invoked, which is after the cli params are mocked out.


This decorator works fine when called in a mesos-cli process such as mesos-cat, but it breaks when called in a test.  That's because the tests, as currently written, import the cli/cmds/* modules before mocking out the cli args.